vtysh/vtysh.c

```diff
@@ -2424,21 +2412,6 @@ static int vtysh_exit(struct vty *vty)
 	if (cnode->parent_node)
 		vty->node = cnode->parent_node;
 
-	if (vty->node == CONFIG_NODE) {
-		bool locked = vty->vtysh_file_locked;
-
-		/* resync in case one of the daemons is somewhere else */
-		vtysh_execute("end");
```

Comment:
I didn't like this code and the hacks it required to make it work; however, are we sure that we dont need it? The idea is that some daemons may be in a node that isn't CONFIG_NODE and by doing an "end" followed by the "config" they will be now.

Reply:
I think the only way to know for sure is to merge and see if someone start complaining. Anyway, if this can happen, then it's a bug that must be fixed, not something that should have a workaround. Locking and unlocking the datastores dozens of times during the config reading is not correct.

Reply:
I tried to go and look at the original bug report from the commit that introduced this pattern, but it's lost to time in the quagga system.  In any event I tend to agree with Igor here.  I'd like to find the situation where this occurrs( if it even does anymore ) and fix that.

Reply:
Ok going to resolve this as handled.